### ISSUE TITLE
Optic History not printing changes

### ISSUE THREAD

**Issue by slnw** [opticdev/optic#2518]
**Describe the bug**
```
optic history openapi.json
# api
### Wed Nov 08 2023
### Tue Nov 07 2023
```

**Expected behavior**
I made changes to the API on both days but they fail to show on the changelog. 
There's no -v to print some type of log in app.

**Details (please complete the following information):**
 - OS and version: Ubuntu 22.04
 - Optic version: 0.50.17
 - NodeJS version: v16.20.2

**Comment by acunniffe** [comment]
Hey @slnw -- can you share anything about the type of changes you made? Are they showing up when you use `optic diff` https://www.useoptic.com/docs/diff-openapi with the commit shas?

**Comment by slnw** [comment]
@acunniffe - maybe because i run with validation=loose it doesn't work with history?
```
optic diff openapi-raw.json --base development --web --validation=loose
api openapi-raw.json
Operations: 13 operations added, 51 changed, 20 removed

GET ...
```

```
optic history openapi-raw.json
# api
### Wed Nov 08 2023
### Tue Nov 07 2023
```

**Comment by acunniffe** [comment]
OK @slnw  that sounds like it could be it. What OpenAPI errors do you get when you don't use `validation=loose` ? Are they fixable?

**Comment by slnw** [comment]
@acunniffe not really, I use an openapi spec generator that creates this errors they will eventually be able to fix it on their end but I'll be happy to use history till/if they do

**Comment by acunniffe** [comment]
Got it. So I think what's happening here is that we're skipping the invalid specs. Since we traverse the entire git history we expect that in some commit there is likely to be an invalid version of the spec. Our code will just skip those versions and compare with the next valid one it finds. 

But in this case, where every version is invalid it looks empty. 

I can see what our team can do about making validation=loose an option for history, but it could lead to some really weird results so we'd have to think hard before turning it on.

**Comment by slnw** [comment]
@acunniffe - i think adding a `--no-validation` flag to history will do it, where by default you'll have validation enabled. adding validation to history without being able to disable it will probably hurt not just my usecase.

In any case validation is not related to history by default. I believe most users use the history on the main branch to generate a changelog to their API, adding validation to it won't help them as if the API file is already there it means it's what they use in prod

**Comment by niclim** [comment]
Hi - I fixed a bug with the `optic history` command which was including commits that didn't have any changes in the spec - which is what you posted in the initial post. Published these changes in `0.53.11`

The history command appears to already use loose validation, so I don't believe that was the issue here.

I'm going to close this issue out first, but if you notice this is still happening with the latest version, feel free to reopen this and if it's possible also upload the current spec version and the spec at the specific date (i.e. the two different specs that should show diffs `optic diff <commit_hash>:openapi.yml openapi.yml`)
